### PR TITLE
Feature/hmr kill switch

### PR DESCRIPTION
**Pull request by nicholasio**
Related Issue/RFC: https://github.com/10up/wp-scaffold/pull/132/
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my change.
- [x] All new and existing tests passed.
- [x] I have added a changeset to my PR. See [**CONTRIBUTING**](https://github.com/10up/10up-toolkit/blob/develop/CONTRIBUTING.md) document for instructions

props @Antonio-Laguna @fabiankaegy

### CODE REVIEW

**Review by Antonio-Laguna — APPROVED** [review]
LGTM!

File: projects/10up-theme/includes/blocks/example-block/markup.php [hunk 1]
```diff
@@ -27,6 +27,8 @@
 ?>
 <div class="<?php echo esc_attr( $args['class_name'] ); ?>">
 	<h2 class="wp-block-example-block__title">
-		<?php echo wp_kses_post( $args['attributes']['customTitle'] ); ?>
+		<?php if ( isset( $args['attributes']['customTitle'] ) ): ?>
```
Comment (fabiankaegy) [inline projects/10up-theme/includes/blocks/example-block/markup.php:30]: @nicholasio why are you making this change here? 🤔 are there any instances where you encountered this being undefined? 

Reading the code it should not be possible to be undefined

Comment (nicholasio) [reply projects/10up-theme/includes/blocks/example-block/markup.php:30]: when I added the block and saved without adding a title there wasn't a default title

File: packages/toolkit/config/fast-refresh.php [hunk 2]
```diff
@@ -40,7 +53,7 @@ function() {
 		$dist_url = $vars['dist_url'];
 
 		if ( ! $dist_url || ! $dist_path ) {
-			trigger_error( '10up-toolkit error: you must defined call TenUpToolkit\set_dist_url_path with the URL and path to dist folfer to get fast refresh to work' );
+			wp_die( '10up-toolkit error: you must defined call TenUpToolkit\set_dist_url_path with the URL and path to dist folfer to get fast refresh to work' );
 			return;
```
Comment (peterwilsoncc) [inline packages/toolkit/config/fast-refresh.php]:
```suggestion
```

Nitpick: can be removed now the function dies.

File: projects/10up-theme/functions.php [hunk 3]
```diff
@@ -14,7 +14,11 @@
 define( 'TENUP_THEME_INC', TENUP_THEME_PATH . 'includes/' );
 define( 'TENUP_THEME_BLOCK_DIR', TENUP_THEME_INC . 'blocks/' );
 
-if ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG && file_exists( __DIR__ . '/dist/fast-refresh.php' ) ) {
+$is_local_env = in_array( wp_get_environment_type(), [ 'local', 'development' ], true );
+$is_local_url = strpos( home_url(), '.test' ) || strpos( home_url(), '.local' );
```
Comment (peterwilsoncc) [inline projects/10up-theme/functions.php:18]: If you only support WP 5.9 or later (or PHP 8.0 or later) then you can use `str_ends_with()` to be more precise -- WP polyfills the function in 5.9 or later.

If you need to support WordPress 5.8 and earlier then cool story, please click resolve conversation. :)

**Review by peterwilsoncc — COMMENTED** [review]
Added a couple of notes because I saw the conversation and can't help myself.

File: projects/10up-theme/functions.php (hunk 3 above)
Comment (nicholasio) [reply projects/10up-theme/functions.php:18]: We don't have clear minimum requirements but I know some projects take longer to update WP so I'm going to be conservative and don't use `str_ends_with`. Thanks for the tip!

File: projects/10up-theme/includes/blocks/example-block/markup.php (hunk 1 above)
Comment (nicholasio) [reply projects/10up-theme/includes/blocks/example-block/markup.php:30]: @fabiankaegy maybe there's a bug elsewhere? anyway this is just a test project, I'm not too concerned about this :)

**Pull request merged**